### PR TITLE
add UT of each suggestion algorithm

### PR DESCRIPTION
**Pull request by ytetra**
add unit test of ranfom algorithm, grid algorithm, and hyperband algorithm

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/237)

### CODE REVIEW

**Comment by YujiOshima** [comment]
@ytetra Thank you for a great job!
/ok-to-test

**Comment by YujiOshima** [comment]
Related #42

File: pkg/suggestion/hyperband_service_test.go [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+			t.Errorf("expected %v, but returned %v", exp_tids[j], rtn_tids[j])
+		}
+		if exp_ts[j].TrialId != rtn_ts[j].TrialId || exp_ts[j].StudyId != rtn_ts[j].StudyId {
+			t.Errorf("expected %v, but returned %v", exp_ts[j], rtn_ts[j])
+		}
+	}
+
+}
+
+func TestPurseSuggestionParameters(t *testing.T) {
```
Comment (YujiOshima) [inline pkg/suggestion/hyperband_service_test.go]: nit: Parse

File: pkg/suggestion/grid_service_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package suggestion
+
+import (
+	"testing"
+)
+
+func TestAllocInt(t *testing.T) {
```
Comment (YujiOshima) [inline pkg/suggestion/grid_service_test.go]: Could you add test for `genGrids`, `parseSuggestParam `, and `setP`?

Comment (YujiOshima) [inline pkg/suggestion/grid_service_test.go]: Could you add test for `genGrids`, `parseSuggestParam `, and `setP`?

**Comment by jlewi** [comment]
@ytetra Thanks for doing this! Any chance you update the PR this week? Would be great to close this out.

**Comment by ytetra** [comment]
@jlewi ok. I finish this PR in this week.

File: pkg/suggestion/hyperband_service_test.go (hunk 1 above)
Comment (ytetra) [reply pkg/suggestion/hyperband_service_test.go]: Thanks,I fix it.

File: pkg/suggestion/grid_service_test.go (hunk 2 above)
Comment (ytetra) [reply pkg/suggestion/grid_service_test.go]: ok. Please wait for a while.

Comment (ytetra) [reply pkg/suggestion/grid_service_test.go]: GenGrid, parseSuggestParam, and setP test added.

**Comment by YujiOshima** [comment]
/lgtm
/approve
@ytetra Thank you!

**Pull request merged**